packages/editor/src/components/plugin-preview-dropdown-item/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * WordPress dependencies
+ */
+import { compose } from '@wordpress/compose';
+import { MenuItem } from '@wordpress/components';
+import { withPluginContext } from '@wordpress/plugins';
+import { ActionItem } from '@wordpress/interface';
+
+/**
+ * Renders a menu item in the Preview dropdown, which can be used as a button or link depending on the props provided.
+ * The text within the component appears as the menu item label.
+ *
+ * @param {Object}                props                                 Component properties.
+ * @param {string}                [props.href]                          When `href` is provided, the menu item is rendered as an anchor instead of a button. It corresponds to the `href` attribute of the anchor.
+ * @param {WPBlockTypeIconRender} [props.icon=inherits from the plugin] The icon to be rendered to the left of the menu item label. Can be a Dashicon slug or an SVG WP element.
+ * @param {Function}              [props.onClick]                       The callback function to be executed when the user clicks the menu item.
+ * @param {...*}                  [props.other]                         Any additional props are passed through to the underlying MenuItem component.
+ *
+ * @example
+ * ```jsx
+ * import { __ } from '@wordpress/i18n';
+ * import { PreviewDropdownMenuItem } from './preview-dropdown-menu-item';
+ * import { external } from '@wordpress/icons';
+ *
+ * function onPreviewClick() {
+ *   // Handle preview action
+ * }
+ *
+ * const ExternalPreviewMenuItem = () => (
+ *   <PreviewDropdownMenuItem
+ *     icon={ external }
+ *     onClick={ onPreviewClick }
+ *   >
+ *     { __( 'Preview in new tab' ) }
+ *   </PreviewDropdownMenuItem>
+ * );
+ * ```
+ *
+ * @return {Component} The rendered menu item component.
+ */
+export default compose(
+	withPluginContext( ( context, ownProps ) => {
+		return {
+			as: ownProps.as ?? MenuItem,
+			icon: ownProps.icon || context.icon,
+			name: 'core/plugin-preview-dropdown-menu',
+		};
+	} )
+)( ActionItem );
```

Comment:
We can probably drop the `withPluginContext`. It's only here for icon context, and as @simison [noted](https://github.com/WordPress/gutenberg/pull/64644#issuecomment-2304810407), there's no need for inheriting icons just yet.

Reply:
The note mentioned above for clarity:

> Hesitant allowing custom icons yet, but not a strong opinion. Mostly thinking that later we want to allow multi-selections with check mark, and it's hard to tell how the icon API then should look like in that scenario.

It's all supported out of the box for the `PluginMoreMenuItem`:

<img width="342" alt="Screenshot 2024-08-28 at 09 04 51" src="https://github.com/user-attachments/assets/bbfdf0d5-e20b-4789-a903-61ff91c4c198">

Whenever, the sidebar is selected, it's reflected in the menu with the checkmark icon ✔️. Let's not limit the options that are supported out of the box like the icon that get inherited for all fills. That's the part that makes the More Menu item behave as a checkbox:

https://github.com/WordPress/gutenberg/blob/71e0fae484c5588b7b5101a3827a5265b653af9f/packages/editor/src/components/plugin-sidebar-more-menu-item/index.js#L56-L63

https://github.com/WordPress/gutenberg/blob/71e0fae484c5588b7b5101a3827a5265b653af9f/packages/interface/src/components/complementary-area-more-menu-item/index.js#L43-L44

Reply:
Then, if we want to keep the icon from the plugin context, let's use the `usePluginContext` hook instead of `compose` and `withPluginContext`.

Reply:
If `usePluginContext` exist, then it would be so much more up to recent React standards 😄